### ISSUE TITLE
Hello + Improvements for JS usage

### ISSUE THREAD

**Issue by steveszc** [NonlinearFruit/Creeds.json#46]
Ahoy 👋 

This repo has been enormously helpful to me as I built https://www.catechesis.app/, so thanks!
(repo here: https://github.com/steveszc/catechesis)

Catechesis is a single-page web app built with Ember.js and written in typescript.
Since I was working in typescript with npm/webpack I ended up copying most of the catechism data and converting it into JS POJOs that could be typed with typescript and imported as ES modules directly in my TS app. I also had to make some additions and alterations to the schema to support things like catechism question commentaries, audio, and prayers.

Long story short, even though I essentially hard-forked the data from this repo, I think I'd benefit a lot from being able to directly depend on this package. And if I was able to depend on it directly I could help maintain it, make improvements, etc. 

But there's a decent amount of work to get the existing json into a format that could be published to NPM and consumed in a project like mine. I'm wondering if you have any interest in pursing this, or if you have any thoughts on how best to enable projects in the JS/npm ecosystem to use the work you've done here.

**Comment by NonlinearFruit** [comment]
Nice! Thanks for the feedback! I added Catechesis to the list of resources in the readme. I see two interesting ideas here. We'll use this issue for the first and #47 for the second.

1. Packaging Creeds.json. I'm definitely open to distributing the public domain resources as a package. It would be awesome to publish these to all sorts of registries (I hadn't thought of that). Two caveats is that we'd want to stay as language neutral as possible (I've love to publish packages to multiple repositories) and I don't have permission to distribute some of the content here, so we'd only be able to publish public domain works.

2. The other idea that is interesting is using Typescript. That could more easily enforce the schema and make writing tests a lot easier 🤔

**Comment by NonlinearFruit** [comment]
> if you have any thoughts on how best to enable projects in the JS/npm ecosystem to use the work you've done here.

@steveszc The tests are now in TypeScript with type definitions ([source](https://github.com/NonlinearFruit/Creeds.json/blob/master/tests/types.ts)), if that is helpful. Were there any specific problems using the data in JS?
